addons/upgrade/to-11.1-remove-unused-sources.pl

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/perl
+
+=head1 NAME
+
+to-11.1-remove-unused-sources
+
+=head1 DESCRIPTION
+
+to-11.1-remove-unused-sources
+
+=cut
+
+use strict;
+use warnings;
+use lib qw(/usr/local/pf/lib);
+use lib qw(/usr/local/pf/lib_perl/lib/perl5);
+use pf::IniFiles;
+use pf::file_paths qw($authentication_config_file);
+use pf::util;
+
+my $file = $authentication_config_file;
+if (@ARGV) {
+    $file = $ARGV[0];
+}
+
+our %typesToDelete = map { $_ => undef  } qw(Twitter Pinterest Mirapay Instagram AuthorizeNet);
+my $cs = pf::IniFiles->new(-file => $file, -allowempty => 1);
+my $update = 0;
+
+for my $section ( grep {/^\S+$/} $cs->Sections() ) {
+    my $type = $cs->val($section, 'type');
+    if (exists $typesToDelete{$type}) {
+        print "Removing $section\n";
+        $cs->DeleteSection($section);
+        for my $group ($cs->GroupMembers($section)) {
+            $cs->DeleteSection($group);
```

Comment:
I don't think this script is handling the removal of the sources in the other configuration sections (like connection profiles)